src/js/index.js

```diff
@@ -0,0 +1,134 @@
+import { ALERT_MESSAGE, OPERATION, DIGIT_NUMBER_MEX_LENGTH } from './constants.js';
+import { $ } from './utils.js';
+
+function App() {
+  // 상태값 정의
+  this.state = {
+    total: '0',
+    digits: [],
+    calculateStep: 0,
+    operators: [],
+  };
+
+  // init 함수
+  this.init = () => {
+    initEventListeners();
+  };
+
+  // 렌더 함수
+  const render = () => {
+    $('#total').textContent = this.state.total;
+  };
+
+  // 계산식 정의
+  const operations = {
+    [OPERATION.PLUS]: (a, b) => Number(a) + Number(b),
+    [OPERATION.MINUS]: (a, b) => Number(a) - Number(b),
+    [OPERATION.MULTIPLICATION]: (a, b) => Number(a) * Number(b),
+    [OPERATION.DIVISION]: (a, b) => Number(a) / Number(b),
+  };
+
+  // 상태값 초기화 함수
+  const resetState = () => {
+    this.state.total = '0';
+    this.state.digits = [];
+    this.state.calculateStep = 0;
+    this.state.operators = [];
+  };
+
+  // 계산하는 함수
+  const calculate = () => {
+    // digit이 1개만 입력됐을경우
+    if (!this.state.digits[1]) {
+      this.state.operators[0] = '';
+      this.state.total = Math.floor(this.state.digits[0]);
+      render();
+      return;
+    }
+
+    const total = this.state.digits.reduce((prev, current, index) => {
+      if (prev !== 0 && this.state.operators[index - 1]) {
+        return operations[this.state.operators[index - 1]](prev, current);
+      }
+
+      return prev + Number(current);
+    }, 0);
+
+    resetState();
+    this.state.digits[0] = total + '';
+    this.state.total = Math.floor(total);
+    render();
+  };
```

Comment:
그리고 
```suggestion
  const calculate = () => {
    const { state } = this;
    // digit이 1개만 입력됐을경우
    if (!state.digits[1]) {
      state.operators[0] = '';
      state.total = Math.floor(this.state.digits[0]);
      return render();
    }

    const total = state.digits.reduce((prev, current, index) => {
      if (prev !== 0 && state.operators[index - 1]) {
        return operations[state.operators[index - 1]](prev, current);
      }

      return prev + Number(current);
    }, 0);

    resetState();
    state.digits[0] = total + '';
    state.total = Math.floor(total);
    render();
  };
```
이렇게 state를 this에서 추출하여 사용하면 더 좋을 것 같아요!

그리고 어차피 state를 변경한 다음에 render가 실행되는거라면,

```js
function setState (newState) {
  this.state = { ...this.state, ...newState };
  render();
}
```
이런 setState 함수를 작성해서 사용하면 더 좋지 않을까요?

Reply:
훨씬 코드가 깔끔해졌네요 감사합니다!